e2e/playwright/pay.spec.ts

```diff
@@ -0,0 +1,359 @@
+      }
+    );
+
+    await gqlAdmin(
+      `
+        mutation InsertPublishedFlow(
+          $publishedFlows: [published_flows_insert_input!]!
+        ) {
+          insert_published_flows(
+            objects: $publishedFlows
+          ) {
+            affected_rows
+          }
+        }
+    `,
+      {
+        publishedFlows: publishedFlows,
+      }
+    );
+
+    await gqlAdmin(
+      `
+        mutation InsertOperations($operations: [operations_insert_input!]!) {
+          insert_operations(objects: $operations) {
+            affected_rows
+          }
+        }
+      `,
+      {
+        operations: [operation],
+      }
+    );
+  });
+
+  test.afterAll(async () => {
+    await gqlAdmin(`
+      mutation CleanDatabase {
+        delete_session_backups(where: {}){
+          affected_rows
+        }
+        delete_analytics_logs(where: {}){
+          affected_rows
+        }
+        delete_analytics(where: {}){
+          affected_rows
+        }
+        delete_published_flows(where: {}){
+          affected_rows
+        }
+        delete_operations(where: {}){
+          affected_rows
+        }
+        delete_flows(where: {}) {
+          affected_rows
+        }
+        delete_teams(where: {}) {
+          affected_rows
+        }
+      }
+    `);
+  });
+
+  test.skip("Should pay and send the application", async ({ page }) => {
+    await navigateToPayComponent({ page, mode: "unpublished" });
+
+    await page.getByText("Pay using GOV.UK Pay").click();
+
+    await page.waitForResponse(
+      (resp) => resp.url().includes("create-send-events") && resp.ok()
+    );
+
+    expect(await page.getByText("Application sent").textContent()).toBeTruthy();
+  });
+
+  test("Should pay within GOV.UK and send the application", async ({
+    page,
+  }) => {
+    await navigateToPayComponent({ page, mode: "preview" });
+
+    await page.getByText("Pay using GOV.UK Pay").click();
+
+    // GOV.UK Pay
+    await page
+      .getByRole("textbox", { name: "Card number" })
+      .fill("4444333322221111");
+    await page.getByLabel("Month").fill("12");
+    await page.getByLabel("Year").fill("2099");
+    await page.getByLabel("Name on card").fill("Test t Test");
+    await page.getByLabel("Card security code", { exact: false }).fill("123");
+
+    await page.getByLabel("Building number or name and street").fill("Test");
+    await page
+      .getByRole("textbox", { name: "Enter address line 2" })
+      .fill("123");
+
+    await page.getByLabel("Town or city").fill("Test");
+    await page.getByLabel("Postcode").fill("HP111BB");
+    await page
+      .getByLabel("Email")
+      .fill("simulate-delivered@notifications.service.gov.uk");
+
+    // GOV.UK Pay submit button is actionable but if there is no delay here the click is ignored
+    await page.waitForTimeout(1000);
+
+    await page.locator("button#submit-card-details").click();
+    await page.getByRole("button", { name: "Confirm payment" }).click();
```

Comment:
It would also be helpful to have a test that captures the journey of clicking "Cancel payment" within Gov UK Pay - which should trigger a state = "failed" & reset on our end (reset meaning you land back on Pay component and Gov Pay inputs are empty when you proceed to pay again).